### DISCUSSION TITLE
Sorting when buffer switching

### DISCUSSION THREAD

**Discussion by danielfleischer** [minad/consult#1011] [Q&A] [answered]
Hi, why wouldn't `consult-buffer` support sorting, especially when used with `orderless` or similar?  This could be a massive qol improvement, at least consider it customizable. 

https://github.com/minad/consult/blob/d84bbc1ea6ae4f85204c22102cbc4e54accbb9c4/consult.el#L4645

**Comment by danielfleischer** [comment]
Compared with `consult-outline`, `consult-line` or `consult-grep`, which have an inherent ordering to results, when you switch buffers and you have files, buffers, bookmarks, you might want ordering that is based on history/usage, e.g. using `vertico`. Setting `:sort t` let you do that or at least offer this option to the user, as `vertico` integration is probably assumed. Results are unordered, even if you have `vertico` set to sort by history, if `:sort nil`. 

EDIT: vertico is doing the sorting, not orderless.

**Comment by minad** [comment, marked as answer]
In order to customize sorting use `(consult-customize consult-buffer :sort t)`. Please let me know if this works. `consult-buffer` disables UI sorting by default since the entries already have an inherent order. Buffers are sorted by recency of usage. Furthermore by disabling sorting we make sure that the order of the sources is preserved, which is usually also desired.

**Reply by danielfleischer** [reply]
Thanks, that solution sets the option and it works. I don't know about buffer sorting, but there is a clear difference when setting this sort=true, for the better, maybe it is my config. Thanks!
